user-console/source/org/pentaho/mantle/client/commands/AdhocRunInBackgroundCommand.java

```diff
@@ -160,6 +175,10 @@ public void onError( Request request, Throwable exception ) {
                 public void onResponseReceived( Request request, Response response ) {
                   if ( response.getStatusCode() != Response.SC_OK ) {
                     AdhocRunInBackgroundCommand.this.onError( response );
+                  } else {
+                    onFinished( JSONParser.
+                      parseStrict( response.getText() ).isObject()
```

Comment:
What happens if your JSON is malformed? (And hey, evil proxy-servers have done sillier things than that before).
Can you catch a broader exception here to guard against that and other problems (like nullpointers because of JSON elements missing)?